### PR TITLE
Integration with Jakarta RESTful Web Services

### PR DESCRIPTION
**Pull request by slinkydeveloper**
#121

### CODE REVIEW

File: http/restful-ws/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+
+public class CloudEventSender {
+
+    public Response sendEvent(WebTarget target, CloudEvent event) {
+        return target
+            .path("postEvent")
+            .request()
+            .buildPost(Entity.entity(event, "application/cloudevents"))
```
Comment (FroMage) [inline http/restful-ws/README.md:81]: This mime type and the following one would make for nice constants in your API.

File: http/restful-ws/pom.xml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>${junit-jupiter.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <!-- Jersey -->
+        <dependency>
+            <groupId>org.glassfish.jersey.test-framework.providers</groupId>
```
Comment (FroMage) [inline http/restful-ws/pom.xml:77]: If you haven't tested it with resteasy, there's no guarantee it works with Quarkus.

File: http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            .visit(visitor);
+    }
+
+    @Override
+    public void filter(ClientRequestContext requestContext) throws IOException {
+        if (Utils.isCloudEventEntity(requestContext.getEntity())) {
+            EventFormat format = EventFormatProvider.getInstance().resolveFormat(requestContext.getMediaType().toString());
+
+            if (format != null) {
+                writeStructured(
```
Comment (FroMage) [inline http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java:136]: Uh, so you write in the filter? That's weird, but OK.

File: http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Optional;
+
+@Provider
+@Consumes(MediaType.WILDCARD)
+@Produces(MediaType.WILDCARD)
+public class CloudEventsProvider implements MessageBodyReader<CloudEvent>, MessageBodyWriter<CloudEvent>, ClientRequestFilter {
```
Comment (FroMage) [inline http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java:52]: Note that in Quarkus you'd have to implement `AsyncMessageBodyWriter` if you ever plan on supporting sending cloud events from the IO thread (for async requests).

Comment (FroMage) [reply http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java:52]: That's a RESTEasy specific API.

Comment (slinkydeveloper) [reply http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java:52]: I think this kind of integration should not live in this project, but maybe in another submodule

File: http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java (hunk 3 above)
Comment (slinkydeveloper) [reply http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java:136]: It is, but it works

File: http/restful-ws/pom.xml (hunk 2 above)
Comment (slinkydeveloper) [reply http/restful-ws/pom.xml:77]: I tried to add both dependencies, but for some reason jersey and resteasy clashes

File: http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java (hunk 3 above)
Comment (slinkydeveloper) [reply http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java:138]: With `MessageBodyWriter` some corner cases aren't working

File: http/restful-ws/pom.xml (hunk 2 above)
Comment (jamezp) [reply http/restful-ws/pom.xml:77]: You could use a profile which could be enabled to make the switch of testing with both easy.

**Comment by fabiojose** [comment]
LGTM

File: http/restful-ws/pom.xml (hunk 2 above)
Comment (slinkydeveloper) [reply http/restful-ws/pom.xml:77]: I'll find out at next PR

**Pull request merged**